src/components/CytoscapeGraph/GraphScore.ts

```diff
@@ -47,24 +47,30 @@ function scoreByOutboundEdges(elements: Readonly<DecoratedGraphElements>): Map<s
   return scoreByEdges(elements, ScoringCriteria.OutboundEdges);
 }
 
+type ScoredNode = DecoratedGraphNodeWrapper & {
+  score: number;
+};
+
 // Adds a score to the node elements based on the criteria(s).
 // Scores are all relative to the other nodes. Criteria
 // can include any source of data but typically looks at
 // data from elements such as edge info.
 export function scoreNodes(
   elements: Readonly<DecoratedGraphElements>,
+  setRankResult?: (result: RankResult) => void,
```

Comment:
I think it'd be simpler to just return an object from this function the same way that `normalizeRanks` now does instead of passing in a setter function. Something like `{elements, rankResult}` or `{elements, upperBound}`.

Reply:
I tried that and for some reason ended up preferring this approach, but feel free to change.